daemon/algod/api/server/lib/middlewares/auth.go

```diff
@@ -96,10 +114,19 @@ func Auth(log logging.Logger, apiToken string) func(echo.HandlerFunc) echo.Handl
 				ctx.Request().URL.Path = newPath
 			}
 
+			matchingAuthIndex := -1
 			// Check the token in constant time
-			if subtle.ConstantTimeCompare(providedToken, apiTokenBytes) == 1 {
+			for i, privAuth := range authTokens {
+				compareResult := subtle.ConstantTimeCompare(providedToken, privAuth.tokenBytes)
+				// if compareResult is 1, we want to update matchingAuthIndex with i. otherwise, we want to keep it as is.
+				matchingAuthIndex = compareResult*i + matchingAuthIndex*(1-compareResult)
+			}
+
+			if matchingAuthIndex >= 0 {
 				// Token was correct, keep serving request
-				return next(ctx)
+				if authTokens[matchingAuthIndex].paths[route] {
```

Comment:
it looks like you'll now get an `"Invalid API Token"` instead of a 404/405 if you request a route that doesn't exist?

Reply:
yes, it's an issue, but it's not there. let me revise that.